### ISSUE TITLE
Relation between site and host

### ISSUE THREAD

**Issue by zhengj2007** [BFO-ontology/bfo-trunk#35]
_From [alanruttenberg@gmail.com](https://code.google.com/u/alanruttenberg@gmail.com/) on May 21, 2012 23:57:43_

How to relate a site to a host? The relationship is mentioned in a couple of elucidations.

_Original issue: http://code.google.com/p/bfo/issues/detail?id=36_

**Comment by zhengj2007** [comment]
_From [dosu...@gmail.com](https://code.google.com/u/102674886352087815907/) on May 22, 2012 03:22:38_

I've been assuming part_of (AKA 'is part of').

The full BFO doc, using anatomy as an example (and FMA as a precedent), makes the case for material entities having immaterial parts.  If I understand the doc correctly, those immaterial parts are Sites.

**Comment by zhengj2007** [comment]
_From [alanruttenberg@gmail.com](https://code.google.com/u/alanruttenberg@gmail.com/) on May 22, 2012 07:48:03_

There are two cases: Interior spaces - cavities, and exterior sites such as the example Barry gives of the environment of a cow feeding. (See attached image). In the latter case such sites can overlap others. So image a modern dairy farm, in which there are separate feeding pens for each cow. What, exactly, is the site part of? Suppose it is part of the pen. Then overlapping spaces and sites being part of their hosts would mean that the pens overlap, which would be counter-intuitive, particularly if the pens are independent artifacts.

In the FMA, I think that the cavities that are considered part of a material entity are so because there is a developmental program in which both are created. In the case of sites that are not created by developmental program - e.g. by fiat, or by the nature of nutrient exchange, or in order to control air traffic, I think the relation would be different.

**Comment by zhengj2007** [comment]
_From [alanruttenberg@gmail.com](https://code.google.com/u/alanruttenberg@gmail.com/) on May 22, 2012 07:48:43_

Oops forgot attachment

**Attachment:** [Sites.PNG](http://code.google.com/p/bfo/issues/detail?id=36#hc3)

**Comment by zhengj2007** [comment]
_From [ifo...@gmail.com](https://code.google.com/u/111675750791183175165/) on May 22, 2012 07:56:44_

The document is already too long, and this is a hard problem to resolve. Hence I suggest keeping this at the level of elucidation, for BFO 2.0.

**Comment by zhengj2007** [comment]
_From [dosu...@gmail.com](https://code.google.com/u/102674886352087815907/) on May 22, 2012 08:21:13_

"ELUCIDATION: a is a site means: a is a three-dimensional immaterial entity that is (partially or wholly) bounded by a material entity or a three-dimensional immaterial part thereof. [034-001]
EXAMPLES: a hole in the interior of a portion of cheese, a rabbit hole, the interior of your bedroom, the Grand Canyon, the Piazza San Marco, an air traffic control region defined in the airspace above an airport, the interior of a kangaroo pouch, your left nostril (a fiat part – the opening – of your left nasal cavity), the lumen of your gut, the hold of a ship, the cockpit of an aircraft, the interior of the trunk of your car, the interior of your refrigerator, the interior of your office, Manhattan Canyon)."

All of this seems clear enough - and I see no reason not to use part_of for all.

This is the first time I've heard anything about developmental processes in this context.  Not even sure this makes sense to me:  Would it discount a cavity formed by cavitation of a solid structure, but include rolling into a tube.  Even if it does make sense, it's rather impractical given the many cases where the precise nature of the developmental process involved will be unknown.  Perhaps you are misinterpreting the reference to 'Bauplan'?

As to the 'environment of a pasturing cow' - the main problem seems to me that more explanation of the intention behind the example is needed for us to have a sensible conversation about it.

**Comment by zhengj2007** [comment]
_From [cmung...@gmail.com](https://code.google.com/u/115850439786749293234/) on May 22, 2012 15:56:50_

"In the FMA, I think that the cavities that are considered part of a material entity are so because there is a developmental program in which both are created" - I agree with David that this is problematic. For example, sometimes a cavity is formed when the barrier between two pre-existing cavities ceases to exist - this may be an independent process from the development of the structures bounding the cavities.

I personally find lumens being part of MEs a little unusual (but this is maybe more of an FMA issue).

**Comment by zhengj2007** [comment]
_From [alanruttenberg@gmail.com](https://code.google.com/u/alanruttenberg@gmail.com/) on May 22, 2012 18:50:21_

A conservative point of view would avoid (yet again) overloading part_of (the new is_a). Let's call it as Varzi has, host_of and then (in another iteration) figure out the consequences for considering host_of a subproperty of has_part.

I don't like the FMA's using part_of either, and I have a nagging feeling, somewhat bolstered by discussion with Tom Bittner, that we will run in to problems using part_of. I'll be reassured by logical proof of consistency, not by one or the other you saying part_of seems fine. Part_of (finally now) comes with some axioms. Show me that the axioms don't lead to contradictions or counterintuitive results for seemingly simple cases (e.g. the two cows, for example, and a dozen more) and I'll feel more confident that part_of is a good choice. 

Note that Reference seems to agree with FMA when it wants, and not when it doesn't want to. So we agree about part_of for host/cavities, but deny their use of bounded_by to relate boundaries to what they bound (with bounded_by not a subproperty of has_part).

**Comment by zhengj2007** [comment]
_From [alanruttenberg@gmail.com](https://code.google.com/u/alanruttenberg@gmail.com/) on May 22, 2012 18:59:33_

"For example, sometimes a cavity is formed when the barrier between two pre-existing cavities ceases to exist - this may be an independent process from the development of the structures bounding the cavities."

Case 1, the disintegration of the barrier is part of normal development: In this case the development process I mean includes that process. The barrier (as you define it) is part of the first set of structures bounding the cavities. It's ceasing to exist, if part of normal development, leads to a new structure that developed from the first. Not sure why you would call this independent (from a developmental point of view).

Case 2, the disappearance of the barrier is accidental. In this case we are talking about pathological anatomy, which FMA doesn't address.

In any case, I think the issue is that there seems to be a recent gravitational field operating on relations, pulling them over the event horizon into part_of. Part_of, already being somewhat inscrutable, gets harder to understand by virtue of these overloadings, and despite there being are axioms of mereology (arguably providing the only meaning part_of has)  never seem to actually be exercised to see whether extensions actually work.

As an engineering practice (and as a conservative ontological practice) it seems a better strategy to be safe than sorry.

**Comment by zhengj2007** [comment]
_From [cmung...@gmail.com](https://code.google.com/u/115850439786749293234/) on May 22, 2012 19:44:39_

Don't blame part_of, I don't think it's inscrutable or overloaded. On the contrary, once you pick your set of mereological axioms it wins the prize for the being the clearest, least problematic relation.

The inscrutability (at least for me) comes with BFO classes such as ME and IE, and how the axioms for these classes involving part_of are constructed. If IEs can be part_of MEs then I have no idea how to start determining where boundaries of MEs lie except without recourse to dodgy geometric simplifications. I think the problems you mention with cow pens above are just the beginning. I'd like to see some of Thom's theories.

But part_of is the innocent party here. The problem is with ME and IE axioms.

**Comment by zhengj2007** [comment]
_From [alanruttenberg@gmail.com](https://code.google.com/u/alanruttenberg@gmail.com/) on May 22, 2012 19:59:55_

The reason I blame part of is that the axioms relating to mereology are formal and not sufficiently tied to properties in the world (IMO). Any relation that satisfies the axioms is a fine part of relation, yet we always get into arguments about what is and isn't part. What I would prefer to see are part relations that are more clearly tied to physical phenomena - ones which can be objectively agreed-upon to hold, and this is why I was leading the direction of developmentally part of for the FMA discussion.

That said, I understand what you mean about them being the least problematic. They are, at least, supposed to work in a certain way, and the way it works has been put to useful purpose.

Bottom line, part_of is woefully underspecified - there are many many relations (considered extensionally) that can satisfy the axioms. On the other hand I think our intuitions about part of cover a relatively small set of those relations - we just don't say precisely which. For material entities our intuitions don't do _too_ badly, but even there there are arguments that arise in anatomy. Mix in the IE and all bets are off, at least with what's been offered in terms of support thus far.

**Comment by zhengj2007** [comment]
_From [dosu...@gmail.com](https://code.google.com/u/102674886352087815907/) on May 23, 2012 02:37:38_

I agree that having IE be part of ME seems a bit odd, but (with the one exception below) I don't see any nasty side effects from this choice, so, in the interests of making some progress, would be happy to stick with it.  I was working from the assumption that we're translating the official BFO2 doc into OWL and this pretty clear states that part_of should apply between sites and their hosts. 

Having thought some more - I think that treating internal spaces as sites and therefore parts is fine.  I also think that the concept of the extended organism is potentially useful (see e.g.  Dawkins - The Extended Phenotype; http://en.wikipedia.org/wiki/The_Extended_Phenotype ).  But treating various chunks of the environment of an organism as parts is asking for trouble.   Given this, where are the limits of the organism?  Do we define them with (presumably) non-rigid criteria such as 'grazing range'?  If so, surely there are many overlapping and constantly changing external spaces that are part of an organism.

**Comment by zhengj2007** [comment]
_From [alanruttenberg@gmail.com](https://code.google.com/u/alanruttenberg@gmail.com/) on May 23, 2012 09:06:38_

Issue 12 has been merged into this issue.

**Cc:** batchelorc@rsc.org

**Comment by zhengj2007** [comment]
_From [alanruttenberg@gmail.com](https://code.google.com/u/alanruttenberg@gmail.com/) on May 23, 2012 09:12:05_

**Status:** Accepted

**Comment by zhengj2007** [comment]
_From [cmung...@gmail.com](https://code.google.com/u/115850439786749293234/) on May 23, 2012 12:34:28_

I suggest we check in sample/test ontologies into svn to properly explore the consequences of each design choice (this is a good idea anyway, a kind of "junit" test suite for bfo). I'd like to see how blood vessels/lumens/blood would be represented.

Extended organisms and phenotypes are useful concepts. There are a number of possible mereological sum combinations of "core organism" + microbes + internal spaces + various external spaces (e.g. OGMS uses convex hull, which is arbitrarily geometric). Do we need to materialize all as classes? The microbe issue is the most pertinent one for an anatomical ontology.

**Comment by zhengj2007** [comment]
_From [alanruttenberg@gmail.com](https://code.google.com/u/alanruttenberg@gmail.com/) on May 23, 2012 16:01:56_

+1 on tests.
The OWL2 working group created an (OWL) data structure for test cases as well as test harness software to run and report on them. See http://owl.semanticweb.org/page/Test_case_ontology The test cases are collected in a (semantic?) mediawiki. Perhaps someone is interested in trying to get a clone up and running for us?

**Comment by zhengj2007** [comment]
_From [alanruttenberg@gmail.com](https://code.google.com/u/alanruttenberg@gmail.com/) on January 28, 2013 10:58:05_

I didn't finish a sentence in the above, and didn't address the issue of host/site. So after posting this message correcting things I will delete the previous:

A discussion with Barry clarifies that a site occupant is located_in a site (continuant-continuant relation). The spatial region of the site is (will be) by definition big enough that the spatial region of the site occupant is part of it. 

Regarding the relation of host to site, part-of has been discussed, but boundary dependence might be a better relation. Barry indicated he would add boundary dependence to the reference.

**Comment by zhengj2007** [comment]
_From [alanruttenberg@gmail.com](https://code.google.com/u/alanruttenberg@gmail.com/) on May 09, 2013 07:12:11_

If the relation is part of then histories must be restricted to material entities, as otherwise the 1:1 relation continuant to process fails due to the possibility of two continuants occupying the same spatial region (e.g.) the site of a puddle and the water filling it. 

Note also that in a discussion with Werner that two continuants could be located in the same place was a surprise and he thought it ought to be modified. 

My preference would be to have a distinct relation at least, and possibly a distinct relation from sites to regions to avoid the distinct continuants having identical regions.

**Labels:** Milestone-bfo2-release
